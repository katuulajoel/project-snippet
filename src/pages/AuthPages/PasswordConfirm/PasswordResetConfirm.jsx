/* eslint-disable react/no-string-refs */
/* eslint-disable react/prop-types */
import React from "react";
import querystring from "querystring";
import connect from "../../../connectors/AuthConnector";

import Error from "../../../components/core/Error";
import Button from "../../../components/core/Button";
import Input from "../../../components/core/input/index";
import Success from "../../../components/core/Success";
import FieldError from "../../../components/core/FieldError";
import PropTypes from "prop-types";
import AuthLayout, {
  AuthStylingLayoutChildren,
} from "../../../layouts/AuthLayout";

class PasswordResetConfirm extends React.Component {
  static defaultProps = {
    newUser: false,
  };

  static propTypes = {
    uid: PropTypes.string.required,
    token: PropTypes.string.required,
    newUser: PropTypes.bool,
  };

  componentDidUpdate(prevProps) {
    if (this.props.Auth.isReset && !prevProps.Auth.isReset) {
      // eslint-disable-next-line react/no-string-refs
      this.refs.reset_confirm_form.reset();

      const { location } = this.props,
        queryParams = querystring.parse(
          (location.search || "").replace("?", "")
        );
      let next = "/dashboard";
      if (queryParams && queryParams.next) {
        next = queryParams.next;
      }
      window.location.href = next;
    }
  }

  onConfirm = (e) => {
    e.preventDefault();
    let { uid, token } = this.props,
      new_password1 = this.refs.new_password1.value.trim(),
      new_password2 = this.refs.new_password2.value.trim();

    if (!new_password1 || !new_password2) {
      return;
    }
    this.props.AuthActions.resetPasswordConfirm({
      uid,
      token,
      new_password1,
      new_password2,
    });
  };

  render() {
    const { Auth, newUser } = this.props,
      queryParams = querystring.parse((location.search || "").replace("?", ""));
    let isNew = newUser || queryParams.new_user;
    return (
      <AuthLayout>
        <AuthStylingLayoutChildren>
          <form
            onSubmit={this.onConfirm}
            name="reset-confirm"
            role="form"
            ref="reset_confirm_form"
          >
            <h3>{isNew ? "Create" : "Reset"} Password</h3>
            {Auth.error.reset_confirm && Auth.error.reset_confirm.token ? (
              <Error message="Invalid token" />
            ) : null}

            {Auth.isReset ? (
              <Success
                message={`Password ${isNew ? "set" : "changed"} successfully`}
              />
            ) : null}

            {Auth.error.reset_confirm &&
            Auth.error.reset_confirm.new_password1 ? (
              <FieldError message={Auth.error.reset_confirm.new_password1} />
            ) : (
              ""
            )}
            <div className="form-group mt-4">
              <label className="control-label">
                {isNew ? null : "New "}Password
                <span
                  style={{
                    color: "#da3451",
                    paddingLeft: "2px",
                  }}
                >
                  *
                </span>
              </label>
              <div>
                <Input
                  label=""
                  type="password"
                  className="form-control"
                  ref="new_password1"
                  required
                  placeholder="Enter password"
                />
              </div>
            </div>
            {Auth.error.reset_confirm &&
            Auth.error.reset_confirm.new_password2 ? (
              <FieldError message={Auth.error.reset_confirm.new_password2} />
            ) : (
              ""
            )}
            <div className="form-group">
              <label className="control-label">
                Confirm {isNew ? null : "New "}Password
                <span
                  style={{
                    color: "#da3451",
                    paddingLeft: "2px",
                  }}
                >
                  *
                </span>
              </label>
              <div>
                <Input
                  label=""
                  type="password"
                  className="form-control"
                  ref="new_password2"
                  required
                  placeholder="Enter password again"
                />
              </div>
            </div>

            <div className="clearfix text-center">
              <Button
                type="submit"
                className="btn-block"
                disabled={Auth.isResetting}
              >
                {isNew ? "Set" : "Change"} Password
              </Button>
            </div>
          </form>
        </AuthStylingLayoutChildren>
      </AuthLayout>
    );
  }
}

export default connect(PasswordResetConfirm);
