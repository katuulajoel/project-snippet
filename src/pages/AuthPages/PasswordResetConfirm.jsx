import React, { useRef, useEffect, useState } from "react";
import querystring from "querystring";
import { useDispatch, useSelector } from "react-redux";

import Error from "../../components/Error";
import Button from "../../components/Button";
import Input from "../../../components/input/index";
import Success from "../../components/Success";
import FieldError from "../../components/FieldError";
import PropTypes from "prop-types";
import AuthLayout, {
  AuthStylingLayoutChildren,
} from "../../layouts/AuthLayout";
import usePrevious from "../../hooks/usePrevious";
import { resetPasswordConfirm } from "../../actions/AuthActions";

const PasswordResetConfirm = (props) => {
  const dispatch = useDispatch();
  const Auth = useSelector(({ Auth }) => Auth);
  const { newUser, uid, token } = props;
  const prevAuthReset = usePrevious(Auth.isReset);
  const formEl = useRef(null);
  const [password1, setPassword1] = useState("");
  const [password2, setPassword2] = useState("");

  const queryParams = querystring.parse(
    (window.location.search || "").replace("?", "")
  );

  useEffect(() => {
    if (Auth.isReset && !prevAuthReset) {
      formEl.current.reset();

      let next = "/dashboard";
      if (queryParams && queryParams.next) {
        next = queryParams.next;
      }
      window.location.href = next;
    }
  });

  const onConfirm = (e) => {
    e.preventDefault();
    if (!password1 || !password2) {
      return;
    }
    dispatch(
      resetPasswordConfirm({
        uid,
        token,
        password1,
        password2,
      })
    );
  };

  const isNew = newUser || queryParams.new_user;

  return (
    <AuthLayout>
      <AuthStylingLayoutChildren>
        <form
          onSubmit={onConfirm}
          name="reset-confirm"
          role="form"
          ref={formEl}
        >
          <h3>{isNew ? "Create" : "Reset"} Password</h3>
          {Auth.error.reset_confirm && Auth.error.reset_confirm.token && (
            <Error message="Invalid token" />
          )}

          {Auth.isReset && (
            <Success
              message={`Password ${isNew ? "set" : "changed"} successfully`}
            />
          )}

          {Auth.error.reset_confirm &&
          Auth.error.reset_confirm.new_password1 ? (
            <FieldError message={Auth.error.reset_confirm.new_password1} />
          ) : (
            ""
          )}
          <div className="form-group mt-4">
            <label className="Auth_label">
              {isNew ? null : "New "}Password
              <span>*</span>
            </label>
            <div>
              <Input
                label=""
                type="password"
                className="form-control"
                required
                placeholder="Enter password"
                value={password1}
                onChange={(e) => setPassword1(e.target.value.trim())}
              />
            </div>
          </div>
          {Auth.error.reset_confirm &&
          Auth.error.reset_confirm.new_password2 ? (
            <FieldError message={Auth.error.reset_confirm.new_password2} />
          ) : (
            ""
          )}
          <div className="form-group">
            <label className="Auth_label">
              Confirm {isNew ? null : "New "}Password
              <span>*</span>
            </label>
            <div>
              <Input
                label=""
                type="password"
                className="form-control"
                required
                placeholder="Enter password again"
                value={password2}
                onChange={(e) => setPassword2(e.target.value.trim())}
              />
            </div>
          </div>

          <div className="clearfix text-center">
            <Button
              type="submit"
              className="btn-block"
              disabled={Auth.isResetting}
            >
              {isNew ? "Set" : "Change"} Password
            </Button>
          </div>
        </form>
      </AuthStylingLayoutChildren>
    </AuthLayout>
  );
};

PasswordResetConfirm.defaultProps = {
  newUser: false,
};

PasswordResetConfirm.propTypes = {
  uid: PropTypes.string.required,
  token: PropTypes.string.required,
  newUser: PropTypes.bool,
  Auth: PropTypes.any,
};

export default PasswordResetConfirm;
