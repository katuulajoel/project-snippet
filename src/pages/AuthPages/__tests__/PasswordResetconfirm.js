import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import PasswordResetConfirm from "../PasswordResetConfirm";
import { mount } from "enzyme/build";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";

const middlewares = [thunk];

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

const mockAppState = {
  Auth: {
    user: { uid: 123, email: "test@gmail.com" },
    isMakingRequest: {},
    errors: {},
  },
};

const mockAppStore = (state) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state || mockAppState);
};

describe("Login snapshot test", () => {
  it("Should match snapshot test", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={mockAppStore()}>
            <PasswordResetConfirm />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call authenticate function", async () => {
    const store = mockAppStore();
    const wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <PasswordResetConfirm />
        </Provider>
      </BrowserRouter>
    );

    wrapper
      .find("#password1")
      .at(1)
      .simulate("change", { target: { value: "password123" } });

    wrapper
      .find("#password2")
      .at(1)
      .simulate("change", { target: { value: "password123" } });

    wrapper.find("button[type='submit']").simulate("submit");

    const expectedActions = [
      {
        type: "RESET_PASSWORD_CONFIRM_START",
        credentials: {
          uid: "",
          token: "",
          password1: "password123",
          password2: "password123",
        },
      },
    ];
    expect(store.getActions()).toEqual(expectedActions);
  });

  it("should not call resetPasswordConfirm function if no username and password", async () => {
    const store = mockAppStore();
    const wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <PasswordResetConfirm />
        </Provider>
      </BrowserRouter>
    );

    wrapper.find("button[type='submit']").simulate("submit");
    expect(store.getActions()).toEqual([]);
  });

  it("should redirect if authenticated", async () => {
    render(
      <MemoryRouter>
        <Provider
          store={mockAppStore({
            Auth: { user: { id: 123456 }, isMakingRequest: {}, errors: {} },
          })}
        >
          <PasswordResetConfirm />
        </Provider>
      </MemoryRouter>
    );
    expect(mockHistoryPush).toHaveBeenCalledWith("/dashboard");
  });
});
