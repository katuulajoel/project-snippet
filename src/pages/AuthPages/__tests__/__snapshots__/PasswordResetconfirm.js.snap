// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot test Should match snapshot test 1`] = `
<div
  className="sc-dlnjwi gkujyx"
>
  <div
    className="grid_layout d-block d-md-grid"
  >
    <div
      className="grid-1 d-flex"
    >
      <div
        className="child__item container"
        data-testid="auth-layout"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          <img
            className="mb-4 logo"
            src={Object {}}
          />
        </a>
        <div
          className="sc-hKFxyN gtgbrS"
        >
          <form
            name="reset-confirm"
            onSubmit={[Function]}
            role="form"
          >
            <h3>
              Reset
               Password
            </h3>
            
            <div
              className="form-group mt-4"
            >
              <label
                className="Auth_label"
              >
                New 
                Password
                <span>
                  *
                </span>
              </label>
              <div>
                <div
                  className="sc-bdnxRM iEBWwR"
                  data-testid="input"
                >
                  <button
                    className="input-icon "
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      fill="none"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 5.41667C11.5323 5.41157 13.0349 5.83899 14.3351 6.64977C15.6353 7.46056 16.6804 8.6218 17.35 10C15.975 12.8083 13.1667 14.5833 10 14.5833C6.83337 14.5833 4.02504 12.8083 2.65004 10C3.31972 8.6218 4.36473 7.46056 5.66494 6.64977C6.96515 5.83899 8.46776 5.41157 10 5.41667ZM10 3.75C5.83337 3.75 2.27504 6.34167 0.833374 10C2.27504 13.6583 5.83337 16.25 10 16.25C14.1667 16.25 17.725 13.6583 19.1667 10C17.725 6.34167 14.1667 3.75 10 3.75ZM10 7.91667C10.5526 7.91667 11.0825 8.13616 11.4732 8.52686C11.8639 8.91756 12.0834 9.44747 12.0834 10C12.0834 10.5525 11.8639 11.0824 11.4732 11.4731C11.0825 11.8638 10.5526 12.0833 10 12.0833C9.44751 12.0833 8.9176 11.8638 8.5269 11.4731C8.1362 11.0824 7.91671 10.5525 7.91671 10C7.91671 9.44747 8.1362 8.91756 8.5269 8.52686C8.9176 8.13616 9.44751 7.91667 10 7.91667ZM10 6.25C7.93337 6.25 6.25004 7.93333 6.25004 10C6.25004 12.0667 7.93337 13.75 10 13.75C12.0667 13.75 13.75 12.0667 13.75 10C13.75 7.93333 12.0667 6.25 10 6.25Z"
                        fill="#8F9BB3"
                      />
                    </svg>
                  </button>
                  <input
                    className="form-control"
                    data-testid=""
                    id="password1"
                    label=""
                    onChange={[Function]}
                    placeholder="Enter password"
                    required={true}
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </div>
            
            <div
              className="form-group"
            >
              <label
                className="Auth_label"
              >
                Confirm 
                New 
                Password
                <span>
                  *
                </span>
              </label>
              <div>
                <div
                  className="sc-bdnxRM iEBWwR"
                  data-testid="input"
                >
                  <button
                    className="input-icon "
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      fill="none"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 5.41667C11.5323 5.41157 13.0349 5.83899 14.3351 6.64977C15.6353 7.46056 16.6804 8.6218 17.35 10C15.975 12.8083 13.1667 14.5833 10 14.5833C6.83337 14.5833 4.02504 12.8083 2.65004 10C3.31972 8.6218 4.36473 7.46056 5.66494 6.64977C6.96515 5.83899 8.46776 5.41157 10 5.41667ZM10 3.75C5.83337 3.75 2.27504 6.34167 0.833374 10C2.27504 13.6583 5.83337 16.25 10 16.25C14.1667 16.25 17.725 13.6583 19.1667 10C17.725 6.34167 14.1667 3.75 10 3.75ZM10 7.91667C10.5526 7.91667 11.0825 8.13616 11.4732 8.52686C11.8639 8.91756 12.0834 9.44747 12.0834 10C12.0834 10.5525 11.8639 11.0824 11.4732 11.4731C11.0825 11.8638 10.5526 12.0833 10 12.0833C9.44751 12.0833 8.9176 11.8638 8.5269 11.4731C8.1362 11.0824 7.91671 10.5525 7.91671 10C7.91671 9.44747 8.1362 8.91756 8.5269 8.52686C8.9176 8.13616 9.44751 7.91667 10 7.91667ZM10 6.25C7.93337 6.25 6.25004 7.93333 6.25004 10C6.25004 12.0667 7.93337 13.75 10 13.75C12.0667 13.75 13.75 12.0667 13.75 10C13.75 7.93333 12.0667 6.25 10 6.25Z"
                        fill="#8F9BB3"
                      />
                    </svg>
                  </button>
                  <input
                    className="form-control"
                    data-testid=""
                    id="password2"
                    label=""
                    onChange={[Function]}
                    placeholder="Enter password again"
                    required={true}
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div
              className="clearfix text-center"
            >
              <button
                className="btn btn-default btn-block  "
                form={null}
                type="submit"
              >
                Change
                 Password
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
    <div
      className="grid-2 d-none d-md-block"
      style={
        Object {
          "backgroundImage": "url([object Object])",
          "backgroundSize": "cover",
        }
      }
    />
  </div>
</div>
`;
